# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLVertexAttribData.h
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGLVertexAttribData.h
--- a/dom/canvas/WebGLVertexAttribData.h	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGLVertexAttribData.h	Wed Jan 08 22:19:23 2020 +0000
@@ -16,24 +16,26 @@
 
 class WebGLVertexAttribData final {
  public:
-  uint32_t mDivisor;
-  bool mEnabled;
+  uint32_t mDivisor = 0;
+  bool mEnabled = false;
 
  private:
-  bool mIntegerFunc;
+  bool mIntegerFunc = false;
 
  public:
   RefPtr<WebGLBuffer> mBuf;
 
  private:
-  GLenum mType;
+  // This and other members below are initialized to dummies here, and later
+  // fully initialized in the ctor.
+  GLenum mType = 0;
   webgl::AttribBaseType mBaseType = webgl::AttribBaseType::Float;
-  uint8_t mSize;  // num of mType vals per vert
-  uint8_t mBytesPerVertex;
-  bool mNormalized;
-  uint32_t mStride;  // bytes
-  uint32_t mExplicitStride;
-  uint64_t mByteOffset;
+  uint8_t mSize = 0;  // num of mType vals per vert
+  uint8_t mBytesPerVertex = 0;
+  bool mNormalized = false;
+  uint32_t mStride = 0;  // bytes
+  uint32_t mExplicitStride = 0;
+  uint64_t mByteOffset = 0;
 
  public:
 #define GETTER(X) \
@@ -52,17 +54,7 @@
 #undef GETTER
 
   // note that these initial values are what GL initializes vertex attribs to
-  WebGLVertexAttribData()
-      : mDivisor(0),
-        mEnabled(false),
-        mIntegerFunc(false),
-        mType(0),
-        mSize(0),
-        mBytesPerVertex(0),
-        mNormalized(false),
-        mStride(0),
-        mExplicitStride(0),
-        mByteOffset(0) {
+  WebGLVertexAttribData() {
     VertexAttribPointer(false, nullptr, 4, LOCAL_GL_FLOAT, false, 0, 0);
   }
 